### ISSUE TITLE
MacOS 10.15.7 - Install - Splash Screen then nothing 

### ISSUE THREAD

**Issue by deanbuxton** [hvdwolf/jExifToolGUI#75]
Hello... I am using a MacOS Pro with v10.15.7  
1. Installed ExifTool - checked in terminal windows, ExifTool works fine by command line. 
2. Downloaded and extracted jExifToolGUI-x86_64-macos-1.6.3-with_jre.dmg.zip 
3. Opened jExifToolGUI.app (Lets MacOS Security open the unapproved file) 
4. I see the application splash screen... then nothing. Just a flash and it closes...  

It does not ask me for the location of ExifTool or anything config wise... 
I don't have a Java 8.1.8 or newer package as I can't find a copy any more. 
So... any suggestions?

**Comment by hvdwolf** [comment]
This is strange. Are you sure you have the  jExifToolGUI-x86_64-macos-1.6.3-**with_jre**.dmg.zip (with java included) or the  jExifToolGUI-x86_64-macos-1.6.3.dmg?
Can you open a terminal? and then do `open /Applications/jExifToolGUI.app` (that is if you installed it in /Applications).
Please let me know if and which errors/warnings you get.

Downloading java can be done from https://adoptopenjdk.net/.
But if you have the jre version, it shouldn't be necessary.

**Comment by deanbuxton** [comment]
Hello HVDWOLD, 
Thx for the reply... Yes I placed the app in /Applications/jExifToolGUI.app. 
I tried terminal just now and ran: open /Applications/jExifToolGUI.app. 
There was a fleeting moment of happiness because the 'splash screen' paused for 1 second. Then it closed the same as executing the app normally. 
I have never seen error or warning at any time and I am running the JRE version of jExifToolGUI for sure. 
I can uninstalling or deleting rather the version I have maybe delete the folder : jexiftoolgui_data as well. 
I see 3 files in that folder if that is what I am supposed to see: 

- cantdisplay.png
- isadg-struct.cfg
- jexiftoolgui.db

I'll let you now in few how it goes... 

Dean

**Comment by deanbuxton** [comment]
Hello - I might be on to something... in the console I have SysPolicyD messages that relate to: 
**com.apple.message.signature: denied:no usable signature**

It almost seems like as soon as MacOS sees a signature it doesn't like, it's killing the process I am starting. 
There is another Finder message : **com.apple.message.quarantined: yes** 

SYSPOLICYD 15:02
assessment granted for jExifToolGUI-x86_64-macos-1.6.3-with_jre.dmg.zip by _XProtect
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:UNBUNDLED
com.apple.message.signature4: 3
com.apple.message.signature3: jExifToolGUI-x86_64-macos-1.6.3-with_jre.dmg.zip
com.apple.message.signature5: UNKNOWN
com.apple.message.signature: granted:_XProtect
SenderMachUUID: 92264DCC-5631-3A5A-A944-28BC718A19DE

SYSPOLICYD 15:02
assessment denied for jExifToolGUI-x86_64-macos-1.6.3-with_jre.dmg
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:UNBUNDLED
com.apple.message.signature3: jExifToolGUI-x86_64-macos-1.6.3-with_jre.dmg
com.apple.message.signature5: UNKNOWN
com.apple.message.signature4: 3
com.apple.message.signature: denied:no usable signature
SenderMachUUID: 92264DCC-5631-3A5A-A944-28BC718A19DE

SYSPOLICYD 15:06
**assessment denied for .app**
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:org.hvdw.jExifToolGUI
com.apple.message.signature3: .app
com.apple.message.signature5: UNKNOWN
com.apple.message.signature4: 1
**com.apple.message.signature: denied:no usable signature**
SenderMachUUID: 92264DCC-5631-3A5A-A944-28BC718A19DE

SYSPOLICYD 15:18
**assessment denied for .app**
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:org.hvdw.jExifToolGUI
com.apple.message.signature3: .app
com.apple.message.signature5: UNKNOWN
com.apple.message.signature4: 1
**com.apple.message.signature: denied:no usable signature**
SenderMachUUID: 92264DCC-5631-3A5A-A944-28BC718A19DE

FINDER
com.apple.message.domain: com.apple.coreservices.applaunch
com.apple.message.__source__: SPI
com.apple.message.path_bucket: 6
com.apple.message.volume_is_network: no
**com.apple.message.quarantined: yes**
com.apple.message.volume_is_removable: yes
com.apple.message.volume_is_root: no
com.apple.message.volume_is_ejectable: yes
com.apple.message.volume_is_disk_image: yes
com.apple.message.bundle_version: 1.6.3
com.apple.message.bundle_identifier: org.hvdw.jExifToolGUI
com.apple.message.bundle_short_version: none
SenderMachUUID: B6873B8C-48E6-3EF7-BAC8-8B5A53C5B276

**Comment by deanbuxton** [comment]
It seems that macOS is behaving in this way and quarantining the process as a defense against malware but in our case, this has to be a bug. The link below suggests the bug has been reported to Apple as ID 50305685 (if this quarantine issue is the same thing that is preventing me from opening jExifToolGUI) It happens on two different MacBook Pro laptops, the 13 inch I have is new and my 15 in about 4 years old, but same OS version - Catalina 10.15.7  

[](https://eclecticlight.co/2019/05/03/serious-flaw-in-macos-quarantine-can-stop-you-from-opening-documents/)

**Comment by mrtngrsbch** [comment]
OSX has no defense mechanism against malware...
but it does have a mechanism against unsigned applications...

is that your case?

**Comment by deanbuxton** [comment]
Not a total fix but this cool: 

With this tool you can see the quarantine header information: 
[](https://eclecticlight.co/xattred-sandstrip-xattr-tools/)
This tool can also allow you to strip the header: XAttred.app 

**com.apple.quarantine (32 bytes):**
{length = 32, bytes = 0x30306332 3b356638 30646465 623b5468 ... 72636869 7665723b }
String content:
00c2;5f80ddeb;The\x20Unarchiver;

**Comment by hvdwolf** [comment]
I really have to dive into this. It is completely new to me.
However: I did add some extra "descriptive tags" to the Info.plist in 1.6.3 which should give more information if you choose options from Finder about properties/details.
 I can hardly imagine that is causing it (I tested on my Catalina), but if so: Can you please download the 1.6.2 and try that one?
The 1.6.2 does not contain that tags yet.
(Info.plist is in `/jExiftoolGUI.app/Contents`)

**Comment by deanbuxton** [comment]
> OSX has no defense mechanism against malware...
> but it does have a mechanism against unsigned applications...
> 
> is that your case?

Yes, that seems to be what I am seeing. It is treating this app as an unsigned app and every time I execute it I can see in the logs that the MacOS it terminating the process. Even in the security console, I am tell it to allow the 'non-app store' app to execute. That seems to get me to where I can launch the program. But when I do, Apple kills it.

**Comment by deanbuxton** [comment]
I tried  1 6 2 version and it. It seems to work the same way... I do see an error in the logs now that show a failed bootstrap path but if I go look in the app, the Java file seems like it is there. The SGKILL message is there as before. 

![Java File is actually there](https://user-images.githubusercontent.com/15960957/95665049-8f2e9680-0b01-11eb-8958-5839c5f24b7c.jpg)
![Step_1_NoBiggie](https://user-images.githubusercontent.com/15960957/95665051-8fc72d00-0b01-11eb-95a6-384d25c13b81.jpg)
![Step2_Allow](https://user-images.githubusercontent.com/15960957/95665052-905fc380-0b01-11eb-819f-f5b9fa4ddd81.jpg)

(com.apple.xpc.launchd.domain.pid.java.2619): **Failed to bootstrap path: path = /Applications/jExifToolGUI.app/Contents/MacOS/jre/Contents/Home/bin/java, error = 2: No such file or directory**


Oct 10 13:53:42 ComputerNamediagnosticd[2610]: System mode client started - Console (2609) - mode: 0xb, filter: "<empty>"
Oct 10 13:53:43 ComputerNamecom.apple.xpc.launchd[1] (com.apple.mdworker.shared.04000000-0200-0000-0000-000000000000[2596]): Service exited due to SIGKILL | sent by mds[118]
Oct 10 13:54:01 ComputerNamebluetoothd[149]: BUG in libdispatch client: read, monitored resource vanished before the source cancel handler was invoked { 0x7ff401424e70[source], ident: 12 / 0xc, handler: 0x108eb5aa0 }
Oct 10 13:54:57 ComputerNamecom.apple.xpc.launchd[1] (com.apple.mdworker.shared.0B000000-0300-0000-0000-000000000000[2608]): Service exited due to SIGKILL | sent by mds[118]
Oct 10 13:54:57 ComputerNamecom.apple.xpc.launchd[1] (com.apple.mdworker.shared.07000000-0700-0000-0000-000000000000[2583]): Service exited due to SIGKILL | sent by mds[118]
Oct 10 13:55:23 ComputerNamecom.apple.xpc.launchd[1] (com.apple.mdworker.shared.04000000-0300-0000-0000-000000000000[2611]): Service exited due to SIGKILL | sent by mds[118]
Oct 10 13:55:54 ComputerNamecom.apple.xpc.launchd[1] (com.apple.mdworker.shared.01000000-0100-0000-0000-000000000000[2612]): Service exited due to SIGKILL | sent by mds[118]
Oct 10 13:56:46 ComputerNamecom.apple.xpc.launchd[1] (com.apple.mdworker.shared.04000000-0400-0000-0000-000000000000[2614]): Service exited due to SIGKILL | sent by mds[118]
Oct 10 13:56:55 ComputerNamecom.apple.xpc.launchd[1] (com.apple.xpc.launchd.domain.pid.java.2619): Failed to bootstrap path: path = /Applications/jExifToolGUI.app/Contents/MacOS/jre/Contents/Home/bin/java, error = 2: No such file or directory

**Comment by hvdwolf** [comment]
I will not change this for the time being. You can still use the app by clicking the "Open anyway" and then it will simply work as of that moment, also with updates.
I will write a note on the Releases page about this and add it to the manual. I actually already knew this, but it simply came not to mind when first answering your issue (my mistake).

I can only overcome this to register as a developer getting a developer ID with Apple (not the same as an Apple ID) and then to codesign the app. Apple wants to know "everything" from me. I have to setup two-factor authentification on at least 2 Apple devices (also different from normal Apple ID), which I do not even have.

And on Apples developer support:
`For software and applications that are downloaded from places other than the Mac App Store, developers can get a Developer ID certificate and submit their software for notarization by Apple. Digitally signing software with a unique Developer ID and including a notarization ticket from Apple lets Gatekeeper verify that the software is not known malware and has not been tampered with. Applications can also take advantage of advanced capabilities such as CloudKit and push notifications.`
What?? So even if I sign my application, they decide whether it's valid or not?? (even that might be even theoretical if my app proves to be safe)
It's against strategies on every other OS whether it is Android, Windows, or whatever. Sure: if you want your app in "their" App store, it should be valid, but outside it, especially with open source software, they should allow "free" software. And it's not only security as Apple is not the most secure OS (3rd place) and the only OS using this.

It is the same protectionist approach that Apple uses for all their hardware and software, and the reason why I switched from Apple to Linux in 2011.

Sorry for the rant about Apple, but to repeat my first sentence:
I will not change this for the time being. You can still use the app by clicking the "Open anyway" and then it will simply work as of that moment, also with updates.

**Comment by deanbuxton** [comment]
Hello... I just wanted to circle back one more time... 
I do choose the allow on the unknown developer 'warning' which is fine..  After I choose to allow it - It tried to open and process is killed by apple every single time. 
I am only prompted for the developer once, and only once when I install a copy or version. 
After that - you run it and it closes... 
The logs that I included is what happens (after) you allow it. The message is repeated - Service exited due to SIGKILL 
It's as if Apple thinks the program is a virus or something.

**Comment by hvdwolf** [comment]
That is strange indeed. I really do not understand why that is happening.
But did you copy the app **again** to `/Applications`, or only once?

The app is tarted by a bootstrap script that starts java with the jar.
From the log I see
```
Oct 10 13:54:01 ComputerNamebluetoothd[149]: BUG in libdispatch client: read, monitored resource vanished before the source cancel handler was invoked { 0x7ff401424e70[source], ident: 12 / 0xc, handler: 0x108eb5aa0 }
...
Oct 10 13:56:55 ComputerNamecom.apple.xpc.launchd[1] (com.apple.xpc.launchd.domain.pid.java.2619): Failed to bootstrap path: path = /Applications/jExifToolGUI.app/Contents/MacOS/jre/Contents/Home/bin/java, error = 2: No such file or directory
```
It seems as if you have an "empty" bundle, because it can't find java inside the bundle.
When mentioning i a virus: Do you have a virus scanner installed? and if so: which one? And can you check its logs if it has something removed or placed in quarantine?
If it signals java as a virus then it has a bug. This java version is definitely not a virus.

**Comment by foxfixer** [comment]
I'm way late in the discussion, but resolved the issue after looking here and returning to the directions.  I've added this comment in the even someone else lands here in search of a solution.

I reinstalled ExifTool, this time being sure to opt for the instructions in the "second item in the Notes section" at exiftool.org/install.html#MacOS. It really helps to control-click on the pkg then select "Open" from the pop-up menu instead of just double-clicking.
